### PR TITLE
Phase2-hgx204 Make a few more bug corrections in view of new V11 geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Partial set of corrections in view of using V11 HGCal geometry

#### PR validation:

A private workflow is created to test this. That workflow will be committed once Kevin's changes to Phase2 scenario is merged

#### if this PR is a backport please specify the original PR:

Not needed

### CODE REVIEW

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -328,10 +328,7 @@ double HGCalDDDConstants::distFromEdgeTrap(double x, double y, double z) const {
   // Take the smaller value
   auto ir = std::lower_bound(hgpar_->radiusLayer_[type].begin(), hgpar_->radiusLayer_[type].end(), r);
   int irad = (int)(ir - hgpar_->radiusLayer_[type].begin());
-  if (irad < hgpar_->iradMinBH_[indx])
-    irad = hgpar_->iradMinBH_[indx];
-  else if (irad > hgpar_->iradMaxBH_[indx])
-    irad = hgpar_->iradMaxBH_[indx];
+  irad = std::min(std::max(irad, hgpar_->iradMinBH_[indx]), hgpar_->iradMaxBH_[indx]);
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: use std::clamp

File: Geometry/HGCalGeometry/test/python/testHGCalWafer_cfg.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("PROD")
+process.load("SimGeneral.HepPDTESSource.pdt_cfi")
+process.load("Configuration.Geometry.GeometryExtended2023D46Reco_cff")
```
Comment (kpedro88) [inline Geometry/HGCalGeometry/test/python/testHGCalWafer_cfg.py]: remember to fix this when D46 is officially added as 2026...

**Comment by kpedro88** [comment]
please test

File: Geometry/HGCalGeometry/test/python/testHGCalWafer_cfg.py (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalGeometry/test/python/testHGCalWafer_cfg.py]: Yes I shall remember

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Done

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by bsunanda** [comment]
@fabiocos Could you kindly integrate it so that I can move forward

**Comment by fabiocos** [comment]
+1

**Pull request merged**